tensorboard/compat/tensorflow_stub/io/gfile_fsspec_test.py

```diff
@@ -0,0 +1,391 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+
+import io
+import os
+
+from tensorboard import test as tb_test
+from tensorboard.compat.tensorflow_stub import errors
+from tensorboard.compat.tensorflow_stub.io import gfile
+
+import fsspec
+
+
+class GFileFSSpecTest(tb_test.TestCase):
+    def get_temp_dir(self):
+        return "file://" + super().get_temp_dir()
+
+    def testExists(self):
+        temp_dir = self.get_temp_dir()
+        self._CreateDeepDirectoryStructure(temp_dir)
+        ckpt_path = os.path.join(temp_dir, "model.ckpt")
+        self.assertTrue(gfile.exists(temp_dir))
+        self.assertTrue(gfile.exists(ckpt_path))
+
+    def testGlob(self):
+        temp_dir = self.get_temp_dir()
+        self._CreateDeepDirectoryStructure(temp_dir)
+        expected = [
+            "foo",
+            "bar",
+            "a.tfevents.1",
+            "model.ckpt",
+            "quuz",
+            "waldo",
+        ]
+        expected_listing = [os.path.join(temp_dir, f) for f in expected]
+        gotten_listing = gfile.glob(os.path.join(temp_dir, "*"))
+        self.assertCountEqual(
+            expected_listing,
+            gotten_listing,
+            "Files must match. Expected %r. Got %r."
+            % (expected_listing, gotten_listing),
+        )
```

Comment:
Rather than checking that the _number_ of entires is the same, can we check that the actual _contents_ of each entry is the same? I want to make sure that each path is well formed.

Reply:
I copied almost all of these test cases from gfile_test.py

Reply:
assertCountEqual counts the occurences of the elements and compares them. So it does check the entries are the same but ignores ordering.